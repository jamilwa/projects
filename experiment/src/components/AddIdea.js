import React, {useEffect, useState} from "react";
import axios from "axios";
import { useHistory } from "react-router-dom";
import Header from "../containers/Header";
export const AddIdea = () => {

    const [ideaTitle, setIdeaTitle] = useState("");
    const [ideaDescription, setIdeaDescription] = useState("");
    const [ideaStorypoints, setIdeaStorypoints] = useState("");


    const history = useHistory();

    const submit = async (e) => {
        e.preventDefault();
        let url = "http://localhost:8090/idea";
        console.log(localStorage.getItem('email')) //to check if the email is decoded or not
        let config = {
            params: {
                email: localStorage.getItem('email'),
            },
            headers: {
                Authorization: "Bearer " + localStorage.getItem('token'),
            }
        }
        let formData = new FormData();
        formData.append("ideaTitle", ideaTitle);
        formData.append("ideaDescription", ideaDescription);
        formData.append("ideaStorypoints", ideaStorypoints);
        
        let respo = await axios.post(url, formData, config);
        
        console.log(respo);
        // history.push("/");
    };


    const onLogoutClicked = () => {
        localStorage.removeItem('token')
        history.push('/login');
      }

  
    return (
        <>
            <form>
            <Header />
            <h1 className="page-title">Add Idea</h1>
                <div className="row justify-content-center">
                    <div className="col-3 center-content">

                        <div className="form-floating mb-3">
                        <label className="form-label">Title</label>
                            <input type="text" className="form-control" name="ideaTitle" placeholder="enter your title"
                                   value={ideaTitle}
                                   onChange={(e) => setIdeaTitle(e.target.value)}/>
                           
                        </div>

                        <div className="form-floating mb-3">
                        <label className="form-label">Description</label>
                            <input type="text" className="form-control" name="ideaDescription" placeholder="Description"
                                   value={ideaDescription}
                                   onChange={(e) => setIdeaDescription(e.target.value)}
                            />
                            
                        </div>
                        <div className="form-floating mb-3">
                        <label className="form-label">Story points</label>
                            <input type="number" className="form-control" name="ideaStorypoints" placeholder="story points"
                                   value={ideaStorypoints}
                                   onChange={(e) => setIdeaStorypoints(e.target.value)}
                            />
                            
                        </div>
                        <hr/>
                        <button onClick={(e) => submit(e)}>Submit</button>
                        <button onClick={()=>onLogoutClicked()}>Logout</button>

                    </div>
                </div>
            </form>
        </>
    );
};
