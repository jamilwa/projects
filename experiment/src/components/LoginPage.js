import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';

export const LoginPage = () => {
    const [errorMessage, setErrorMessage] = useState('');
    const [username, setUsername] = useState('');
    const [password, setPassword] =useState('');
    const [isValid, setIsValid] = useState(false);
    const [message, setMessage] = useState('');

    const history = useHistory();

    const regEx1 = /[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,8}(.[a-z{2,8}])?/g;

    const usernamelValidation = (e) => {
        setUsername(e.target.value);
        if (regEx1.test(username)) {
            setIsValid(true);
          setMessage("");
        } else {
            setIsValid(false);
          setMessage("Enter a valid username.");
        }
      };


    const onLoginClicked = async () => {
        await axios.post('http://localhost:8090/login', {
        username: username,
        password: password
      })
      .then(res => {
        //console.log(res.data.token)
        if(res.data.token){
          localStorage.setItem('token', res.data.token)
          localStorage.setItem('email', username)
          console.log(localStorage.getItem('email'));
        }
        console.log(localStorage.getItem('token'))
        history.push('/add')
            })
    .catch(function (error) {
        setErrorMessage("Please check your entries and try again!");
    });
      
    }

    return(
        <div className="content-container">
            <h1>Login</h1>
            {errorMessage && <div className="fail" style={{color: "red"}}>{errorMessage}</div>}
            <input
                value={username}
                onChange={usernamelValidation}
                placeholder="username" />
                <div className={`message ${isValid ? 'success' : 'error'}`}>
                    {message}
                </div>
            <input
                type="password"
                value={password}
                onChange={e => setPassword(e.target.value)}
                placeholder="password" />
                <button
                    disabled={!username || !password || !isValid}
                    onClick={()=>onLoginClicked()}>Log In</button>
                    <hr />
                <button onClick={() => history.push('/registration')}>Don't have an account? Register here</button>
                

        </div>
    )
}