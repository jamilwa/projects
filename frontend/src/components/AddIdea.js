import React, {useEffect, useState} from "react";
import axios from "axios";
import { useHistory } from "react-router-dom";
import Header from "../containers/Header";
export const AddIdea = () => {

    const [ideaTitle, setIdeaTitle] = useState("");
    const [ideaDescription, setIdeaDescription] = useState("");
    const [ideaStorypoints, setIdeaStorypoints] = useState("");


    const history = useHistory();

    const submit = async (e) => {
        e.preventDefault();
        let url = "http://localhost:8090/idea";
        
        console.log(localStorage.getItem('email')) //to check if the email is decoded or not
        let config = {
            params: {
                email: localStorage.getItem('email'),
            },
            headers: {
                Authorization: "Bearer " + localStorage.getItem('token'),
                Accept: "application/json"
            }
        }

        let obj = {
            "ideaTitle": ideaTitle,
            "ideaDescription": ideaDescription,
            "ideaStorypoints": ideaStorypoints
        }

    
        let respo = await axios.post(url, obj, config);
        console.log(respo);
        history.push("/idealisting");
    };
  
    return (
        <>
            <form>
            <Header />
                <div className="row justify-content-center">
                    <div className="col-3 center-content">

                        <div className="form-floating mb-3">
                        <label className="form-label">Title</label>
                            <input type="text" className="form-control" name="ideaTitle" placeholder="enter your title"
                                   value={ideaTitle}
                                   onChange={(e) => setIdeaTitle(e.target.value)}/>
                           
                        </div>

                        <div className="form-floating mb-3">
                        <label className="form-label">Description</label>
                            <input type="text" className="form-control" name="ideaDescription" placeholder="Description"
                                   value={ideaDescription}
                                   onChange={(e) => setIdeaDescription(e.target.value)}
                            />
                            
                        </div>
                        <div className="form-floating mb-3">
                        <label className="form-label">Story points</label>
                            <input type="number" className="form-control" name="ideaStorypoints" placeholder="story points"
                                   value={ideaStorypoints}
                                   onChange={(e) => setIdeaStorypoints(e.target.value)}
                            />
                            
                        </div>
                        <hr/>
                        <button 
                        disabled={!ideaTitle || !ideaDescription || !ideaStorypoints}
                        className="ui animated button" tableindex="0" onClick={(e) => submit(e)}>
                        <div className="visible content">Submit</div>
                        <div className="hidden content">
                            <i className="chevron circle right icon"></i></div></button>

                            <button 
                        className="ui animated button" tableindex="0" onClick={(e) => history.push('/idealisting')}>
                        <div className="visible content">Back</div>
                        <div className="hidden content">
                            <i className="chevron circle left icon"></i></div></button>


                    </div>
                </div>
            </form>
        </>
    );
};