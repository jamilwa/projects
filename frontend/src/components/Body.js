import React from 'react'

export default function Body() {
    this.state({
        token : null
    })
    this.setState({
        token : localStorage.getItem('token')
    })
   
  return (
    <div>Body {this.state.token}</div>

  )
}
