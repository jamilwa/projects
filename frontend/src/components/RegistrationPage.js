import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import validator from 'validator';
import axios from 'axios';

export const RegistrationPage = () => {
    const [errorMessage, setErrorMessage] = useState('');
    
    const [email, setEmail] = useState('');
    const [password, setPassword] =useState('');
    const [confirmPassword, setConfirmPassword] =useState('');
    const [isValid, setIsValid] = useState(false);
    const [isValid1, setIsValid1] = useState(false);
    const [isValid2, setIsValid2] = useState(false);
    const [message, setMessage] = useState('');
    const [message1, setMessage1] = useState('');
    const [message2, setMessage2] = useState('');
    const history = useHistory();

    const regEx = /[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,8}(.[a-z{2,8}])?/g;

    const emailValidation = (e) => {
        e.preventDefault();
        setEmail(e.target.value);
        if (regEx.test(email)) {
            setIsValid(true);
          setMessage("Email is valid.");
        } else {
            setIsValid(false);
          setMessage("Please enter a valid email.");
        }
      };

      const passwordValidation = (e) => {
          e.preventDefault();
          setPassword(e.target.value);
          if(validator.isStrongPassword(password, {
              minLength: 8, minLowercase: 1,
              minUppercase: 1, minNumbers: 1, minSymbols: 1
          })) {
              setIsValid1(true);
              setMessage1('This is a strong password.')
          } else {
              setIsValid1(false);
              setMessage1('Please choose a stronger password.')
          }
      };

      const confirmPasswordValidation = (e) => {
        setConfirmPassword(e.target.value);
        if(validator.isStrongPassword(confirmPassword, {
            minLength: 8, minLowercase: 1,
            minUppercase: 1, minNumbers: 1, minSymbols: 1
        })) {
            setIsValid2(true);
            setMessage2('')
        } else {
            setIsValid2(false);
            setMessage2('')
        }
    };

    const onRegisterClicked = async () => {
        await axios.post('http://localhost:8090/user', {
            email: email,
            password: password
      })
      history.push('/login');
      console.log('Success');
    }

    return(

        <div className="content-container">
            <h1>Sign Up</h1>
            {errorMessage && <div className="fail">{errorMessage}</div>}
            <input
                value={email}
                onChange={emailValidation}
                placeholder="enter your email" />

                <div className={`message ${isValid ? 'success' : 'error'}`}>
                    {message}
                    </div>
            <input
                type="password"
                value={password}
                onChange={passwordValidation}
                placeholder="enter your password" />

                <div className={`message ${isValid1 ? 'success' : 'error'}`}>
                    {message1}
                    </div>

            <input
                type="password"
                value={confirmPassword}
                onChange={confirmPasswordValidation}
                placeholder="confirm your password" />

                <div className={`message ${isValid2 ? 'success' : 'error'}`}>
                    {message2}
                    </div>

                <hr />
                <button
                    disabled={!password || !isValid || !isValid1 || !isValid2 || password !== confirmPassword}
                    onClick={()=>onRegisterClicked()}>Register</button>

                <hr />
                <button onClick={() => history.push('/login')}>Already have an account? Log in here</button>


        </div>
    )
}