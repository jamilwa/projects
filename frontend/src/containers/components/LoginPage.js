import React from 'react';
import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';

export const LoginPage = () => {
    
    const [errorMessage, setErrorMessage] = useState('');
    const [username, setUsername] = useState('');
    const [password, setPassword] =useState('');
    const [isValid, setIsValid] = useState(false);
    const [message, setMessage] = useState('');

    const history = useHistory();

    const regEx1 = /[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,8}(.[a-z{2,8}])?/g;
    function checkLogin() {
        if(localStorage.getItem('email')){
            history.push('/idealisting')
        }
    }
    checkLogin();

    const usernamelValidation = (e) => {
       
        setUsername(e.target.value);
        if (regEx1.test(username)) {
            setIsValid(true);
          setMessage("");
        } else {
            setIsValid(false);
          setMessage("Enter a valid username.");
        }
      };


    const onLoginClicked = async () => {
        await axios.post('http://localhost:8090/login', {
        username: username,
        password: password
      })
      .then(res => {

        if(res.data.token){
          localStorage.setItem('token', res.data.token)
          localStorage.setItem('email', username)
          console.log(localStorage.getItem('email'));
        }
        console.log(localStorage.getItem('token'))
        history.push('/idealisting')
            })
    // .catch(function (error) {
    //     setErrorMessage("Please check your entries and try again!", error);
    // });
        .catch(err => {
            setErrorMessage('Error Occured! ' + err);
            localStorage.clear();
        })
      
    }

    return(
        <div className="content-container">
            <h1 style={{color:"green", fontSize: "22px", textAlign: "center", fontStyle: "-moz-initial", fontVariant: "all-petite-caps"}}>Log-in to your account</h1>
            {errorMessage && <div className="fail" style={{color: "red"}}>{errorMessage}</div>}
            <input
                id='input-username'
                value={username}
                onChange={usernamelValidation}
                className='user icon'
                placeholder="username" />
                <div className={`message ${isValid ? 'success' : 'error'}`}>
                    {message}
                </div>
            <input
                id='input-password'
                type="password"
                value={password}
                onChange={e =>setPassword(e.target.value)}
                placeholder="password" />

                <button id='btn-primary' className="ui animated button" tableindex="0" 
                    disabled={!username || !password || !isValid}
                    onClick={()=>onLoginClicked()}>
                    <div className="visible content">Log In</div>
                    <div className="hidden content">
                    <i className="sign in alternate icon"></i></div>
                </button>
                    <hr />
                <button className="ui animated button" tableindex="0" onClick={() => history.push('/registration')}>
                <div className="visible content">Don't have an account? Register here</div>
                <div className="hidden content">
                    <i className="chevron circle left icon"></i></div>
                </button>
                

        </div>
    )
}

