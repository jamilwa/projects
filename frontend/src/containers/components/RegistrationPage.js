import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import validator from 'validator';
import axios from 'axios';

export const RegistrationPage = () => {
    const [errorMessage, setErrorMessage] = useState('');
    
    const [email, setEmail] = useState('');
    const [password, setPassword] =useState('');
    const [confirmPassword, setConfirmPassword] =useState('');
    const [isValid, setIsValid] = useState(false);
    const [isValid1, setIsValid1] = useState(false);
    const [isValid2, setIsValid2] = useState(false);
    const [message, setMessage] = useState('');
    const [message1, setMessage1] = useState('');
    const [message2, setMessage2] = useState('');
    const history = useHistory();

    const regEx = /[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,8}(.[a-z{2,8}])?/g;

    const emailValidation = (e) => {
        e.preventDefault();
        setEmail(e.target.value);
        if (regEx.test(email)) {
            setIsValid(true);
          setMessage("Email is valid.");
        } else {
            setIsValid(false);
          setMessage("Please enter a valid email.");
        }
      };

      const passwordValidation = (e) => {
          e.preventDefault();
          setPassword(e.target.value);
          if(validator.isStrongPassword(password, {
              minLength: 8, minLowercase: 1,
              minUppercase: 1, minNumbers: 1, minSymbols: 1
          })) {
              setIsValid1(true);
              setMessage1('This is a strong password.')
          } else {
              setIsValid1(false);
              setMessage1('Minimum 8 characters, an Upper case leter, a numeral and a symbol.')
          }
      };

      const confirmPasswordValidation = (e) => {
        setConfirmPassword(e.target.value);
        if(validator.isStrongPassword(confirmPassword, {
            minLength: 8, minLowercase: 1,
            minUppercase: 1, minNumbers: 1, minSymbols: 1
        })) {
            setIsValid2(true);
            setMessage2('')
        } else {
            setIsValid2(false);
            setMessage2('')
        }
    };

    const onRegisterClicked = async () => {
        await axios.post('http://localhost:8080/user', {
            email: email,
            password: password
      })
      .then(resp => {
        history.push('/');
        console.log('Success', resp);
      })
      .catch(function (error) {
        setErrorMessage("User is already registered!", error);
    });
      
    }

    return(

        <div className="content-container">
            <h1 style={{color:"green", fontSize: "23px", textAlign: "center", fontStyle: "-moz-initial", fontVariant: "all-petite-caps"}}>Create an account</h1>
                {errorMessage && <div className="fail" style={{color: "red"}}>{errorMessage}</div>}
            <input
                id="userName-input"
                value={email}
                onChange={emailValidation}
                placeholder="enter your email" />

                <div className={`message ${isValid ? 'success' : 'error'}`}>
                    {message}
                </div>

            <input
                type="password"
                value={password}
                onChange={passwordValidation}
                placeholder="enter your password" />

                <div className={`message ${isValid1 ? 'success' : 'error'}`}>
                    {message1}
                </div>

            <input
                type="password"
                value={confirmPassword}
                onChange={confirmPasswordValidation}
                placeholder="confirm your password" />

                <div className={`message ${isValid2 ? 'success' : 'error'}`}>
                    {message2}
                </div>

                <hr />
                <button id="btn-register" className="ui animated button" tableindex="0"
                    disabled={!password || !isValid || !isValid1 || !isValid2 || password !== confirmPassword}
                    onClick={()=>onRegisterClicked()}>
                    <div className="visible content">Register</div>
                    <div className="hidden content">
                        <i className="chevron right icon"></i>
                    </div>
                </button>

                <hr />
                <button className="ui animated button" tableindex="0" onClick={() => history.push('/')}>
                    <div className="visible content">Already have an account? Log in here</div>
                    <div className="hidden content">
                        <i className="chevron circle right icon"></i>
                    </div>
                </button>

        </div>
    )
}
