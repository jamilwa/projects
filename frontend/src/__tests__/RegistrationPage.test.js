import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import { configure ,shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { RegistrationPage } from "../containers/components/RegistrationPage";
import "core-js/stable";
import "regenerator-runtime/runtime";

configure({ adapter: new Adapter() });

it("check mount",()=> {
    const renderer = shallow(<RegistrationPage />);
    renderer.find('#btn-register').simulate('click');
    expect(renderer).toMatchSnapshot();
})
// it("check mounts",()=> {
//     const renderer = shallow(<RegistrationPage/>);
//     //renderer.find('#btn-clear').simulate('click');
//     // expect(renderer).toMatchSnapshot();
// })


// it("check for test cases ya",()=> {
//     const renderer = shallow(<RegistrationPage/>);
//     // renderer.instance().setState({
//     //     loggedIn:true
//     // })
//     // expect(renderer).toMatchSnapshot();
// })

it("check for test cases",()=> {
    const renderer = new ShallowRenderer();
    renderer.render(<RegistrationPage />);
    expect(renderer).toMatchSnapshot();
})