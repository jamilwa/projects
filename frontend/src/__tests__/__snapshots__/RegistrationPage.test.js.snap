// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check for test cases 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useDeferredValue": [Function],
    "useEffect": [Function],
    "useImperativeHandle": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useResponder": [Function],
    "useState": [Function],
    "useTransition": [Function],
  },
  "_element": <RegistrationPage />,
  "_firstWorkInProgressHook": Object {
    "memoizedState": "",
    "next": Object {
      "memoizedState": "",
      "next": Object {
        "memoizedState": "",
        "next": Object {
          "memoizedState": "",
          "next": Object {
            "memoizedState": false,
            "next": Object {
              "memoizedState": false,
              "next": Object {
                "memoizedState": false,
                "next": Object {
                  "memoizedState": "",
                  "next": Object {
                    "memoizedState": "",
                    "next": Object {
                      "memoizedState": "",
                      "next": null,
                      "queue": Object {
                        "dispatch": [Function],
                        "first": null,
                      },
                    },
                    "queue": Object {
                      "dispatch": [Function],
                      "first": null,
                    },
                  },
                  "queue": Object {
                    "dispatch": [Function],
                    "first": null,
                  },
                },
                "queue": Object {
                  "dispatch": [Function],
                  "first": null,
                },
              },
              "queue": Object {
                "dispatch": [Function],
                "first": null,
              },
            },
            "queue": Object {
              "dispatch": [Function],
              "first": null,
            },
          },
          "queue": Object {
            "dispatch": [Function],
            "first": null,
          },
        },
        "queue": Object {
          "dispatch": [Function],
          "first": null,
        },
      },
      "queue": Object {
        "dispatch": [Function],
        "first": null,
      },
    },
    "queue": Object {
      "dispatch": [Function],
      "first": null,
    },
  },
  "_forcedUpdate": false,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": <div
    className="content-container"
  >
    <h1
      style={
        Object {
          "color": "green",
          "fontSize": "23px",
          "fontStyle": "-moz-initial",
          "fontVariant": "all-petite-caps",
          "textAlign": "center",
        }
      }
    >
      Create an account
    </h1>
    
    <input
      id="userName-input"
      onChange={[Function]}
      placeholder="enter your email"
      value=""
    />
    <div
      className="message error"
    >
      
    </div>
    <input
      onChange={[Function]}
      placeholder="enter your password"
      type="password"
      value=""
    />
    <div
      className="message error"
    >
      
    </div>
    <input
      onChange={[Function]}
      placeholder="confirm your password"
      type="password"
      value=""
    />
    <div
      className="message error"
    >
      
    </div>
    <hr />
    <button
      className="ui animated button"
      disabled={true}
      id="btn-register"
      onClick={[Function]}
      tableindex="0"
    >
      <div
        className="visible content"
      >
        Register
      </div>
      <div
        className="hidden content"
      >
        <i
          className="chevron right icon"
        />
      </div>
    </button>
    <hr />
    <button
      className="ui animated button"
      onClick={[Function]}
      tableindex="0"
    >
      <div
        className="visible content"
      >
        Already have an account? Log in here
      </div>
      <div
        className="hidden content"
      >
        <i
          className="chevron circle right icon"
        />
      </div>
    </button>
  </div>,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;

exports[`check mount 1`] = `ShallowWrapper {}`;
